verification/src/solidity/compiler_fetcher.rs

```diff
@@ -0,0 +1,326 @@
+use crate::compiler::{fetcher::Fetcher, version::CompilerVersion};
+use async_trait::async_trait;
+use primitive_types::H256;
+use serde::Deserialize;
+use std::{
+    collections::HashMap,
+    fs::{File, OpenOptions},
+    io::ErrorKind,
+    os::unix::prelude::OpenOptionsExt,
+    path::{Path, PathBuf},
+};
+use thiserror::Error;
+use url::Url;
+
+#[derive(Error, Debug)]
+pub enum ListError {
+    #[error("fetching list json returned error: {0}")]
+    ListJsonFetch(reqwest::Error),
+    #[error("cannot parse list json file: {0}")]
+    ParseListJson(reqwest::Error),
+    #[error("error parsing 'path' field: {0}")]
+    Path(url::ParseError),
+}
+
+#[derive(Default)]
+pub struct Releases(HashMap<CompilerVersion, CompilerInfo>);
```

Comment:
Isn't `Release` too similar with `ReleaseVersion` compilers? I cannot come up with anything better, but maybe there is still a name that would be less associated with non-nightly versions?

Reply:
thought about it too but there was `releases` variable in github fetcher before, so decided to leave this name.

maybe `Compilers` / `CompilerVersions` / `Versions` ?

Reply:
I prefer `Versions`, but if you don't agree, I can reset last [commit](https://github.com/blockscout/blockscout-rs/pull/41/commits/4a0133377df4b628e173ed445bb374f4e22c9b01)